apps/api/src/database/repositories/competition-repository.ts

```diff
@@ -429,6 +429,53 @@ export async function getAgentPortfolioSnapshots(
   }
 }
 
+/**
+ * Get agent's ranking in a specific competition
+ * @param agentId Agent ID
+ * @param competitionId Competition ID
+ * @returns Object with rank and totalAgents
+ */
+export async function getAgentCompetitionRanking(
+  agentId: string,
+  competitionId: string,
+): Promise<{ rank: number; totalAgents: number }> {
+  try {
+    // Get all latest portfolio snapshots for the competition
+    const snapshots = await getLatestPortfolioSnapshots(competitionId);
+
+    if (snapshots.length === 0) {
+      return { rank: 0, totalAgents: 0 };
```

Comment:
oh and same thing here re: the point below....if there's no `bestPlacement`, then we should return `undefined` because handling "zero rank" can potentially be misleading for the client to handle.